src/core/stream_send.c

```diff
@@ -1537,8 +1532,21 @@ QuicStreamOnAck(
             "[strm][%p] Send State: %hhu",
             Stream,
             QuicStreamSendGetState(Stream));
-        QuicStreamCleanupReliableReset(Stream);
-        QuicStreamSendShutdown(Stream, FALSE, TRUE, FALSE, Stream->SendShutdownErrorCode);
+        //
+        // Set the flags and clear any outstanding send path frames.
+        //
+        Stream->Flags.LocalCloseAcked = TRUE;
+        QuicSendClearStreamSendFlag(
+            &Stream->Connection->Send,
+            Stream,
+            QUIC_STREAM_SEND_FLAG_DATA_BLOCKED |
+            QUIC_STREAM_SEND_FLAG_RELIABLE_ABORT |
+            QUIC_STREAM_SEND_FLAG_DATA |
+            QUIC_STREAM_SEND_FLAG_OPEN |
+            QUIC_STREAM_SEND_FLAG_FIN);
```

Comment:
Do we already have a #define for these? If not, perhaps we should define something that represents this logic grouping.

Reply:
Done! 91f4021db5f2d340a613e505675611cb83c29e18